ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -733,16 +1147,29 @@ def _checkConvObjective(self, traj, **kwargs):
       @ In, kwargs, dict, dictionary of parameters for convergence criteria
       @ Out, converged, bool, convergence state
     """
+    # _optPointHistory is used to check that we run the algorithm one step
     if len(self._optPointHistory[traj]) < 2:
       return False
-    o1, _ = self._optPointHistory[traj][-1]
-    obj = o1[self._objectiveVar]
-    converged = (obj == self._convergenceCriteria['objective'])
-    self.raiseADebug(self.convFormat.format(name='objective',
-                                            conv=str(converged),
-                                            got=obj,
-                                            req=self._convergenceCriteria['objective']))
-
+    # An alternative was to use:
+    # o1, _ = self._optPointHistory[traj][-1]
+    # but that will only search the "best points" so is slower at finding
+    # one that matches the objective.
+    o1 = kwargs['new']
+    #print(f"{o1=}")
+    for j in range(len(np.atleast_1d(o1[self._objectiveVar[0]]))):
+      converged = True
+      bestObjective = []
+      for i,objVar in enumerate(self._objectiveVar):
+        #print(f"{objVar=}  {np.atleast_1d(o1[objVar])[j]=} {self._convergenceCriteria['objective'][i]=}")
+        currentObj = np.atleast_1d(o1[objVar])[j]*self._objMult[objVar]
+        bestObjective.append(currentObj*self._objMult[objVar])
+        converged = (currentObj == self._convergenceCriteria['objective'][i]) and converged
+      if converged:
+        if self._isMultiObjective:
+          self.multiBestObjective = np.array([bestObjective])
+        else:
+          self.bestObjective = bestObjective[0]
```

Comment:
I would suggest we do not define different variables, I think we can use the same data structure to handle or store the best objective values.

Reply:
Fixed in 19d6cf4ed1d5e9bb9c8ab565a25012ec2764b00b